lib/pluginmanager/install.rb

```diff
@@ -175,6 +176,26 @@ def plugins_gems
     version ? [plugins_arg << version] : plugins_arg.map { |plugin| [plugin, nil] }
   end
 
+  def local_gem?
+    plugins_arg.any? { |plugin_arg| LogStash::PluginManager.plugin_file?(plugin_arg) }
+  end
+
+  def update_logstash_mixin_dependencies(install_list)
+    return if !verify? || preserve? || development? || local? || local_gem?
+
+    puts "Resolving mixin dependencies"
+    LogStash::Bundler.setup!
+    plugins = LogStash::Bundler.consolidate_dependencies(install_list.map { |arr| arr[0] }, false)
```

Comment:
If you take my advice of making the caller responsible for omitting, then this becomes something like:

~~~ ruby
    plugins_to_update = install_list.map(&:first)
    unlock_dependencies = LogStash::Bundler.consolidate_dependencies(plugins_to_update) - plugins_to_update

    if unlock_dependencies.any?
      puts "Updating mixin dependencies #{unlock_dependencies.join(', ')}"
      options = {:update => unlock_dependencies, :rubygems_source => gemfile.gemset.sources}
      LogStash::Bundler.invoke!(options)
    end
~~~